caching-service/src/main/java/org/zowe/apiml/caching/service/infinispan/config/InfinispanConfig.java

```diff
@@ -71,6 +69,21 @@ void updateKeyring() {
         }
     }
 
+    static String getRootFolder() {
+        // using getenv().get is because of system compatibility (see non-case sensitive on Windows)
+        String instanceId = System.getenv().get("ZWE_haInstance_id");
+        if (StringUtils.isBlank(instanceId)) {
+            instanceId = "localhost";
+        }
+
+        String workspaceFolder = System.getenv().get("ZWE_zowe_workspaceDirectory");
+        if (StringUtils.isBlank(workspaceFolder)) {
+            return Paths.get("caching-service", instanceId).toString();
```

Comment:
is there some value that would produce exception?

Reply:
InvalidPathException – If the path string cannot be converted

It would lead to an error during the start-up if the instanceId contains unexpected characters. I would fail same as the `start.sh` script in the ZAAS, see `-Dapiml.cache.storage.location=${ZWE_zowe_workspaceDirectory}/api-mediation/${ZWE_haInstance_id:-localhost} \`